infrastructure/http/DefaultsExtractor.go

```diff
@@ -0,0 +1,53 @@
+package http
+
+import nethttp "net/http"
+
+type defaultsExtractor struct {
+}
+
+func (self defaultsExtractor) Extract(input map[string]interface{}) HttpActionState {
```

Comment:
Also we should decide on what the 'instance' name should be and update all code for consistency... we currently have (that I am aware of) `instance`, `i`, `this`, `self` plus others.